### PR TITLE
Config docs

### PR DESCRIPTION
**Pull request by purnimagupta**
This PR add examples for configuring gdeploy for various use cases listed below:
1. Creating a 1x3 GlusterFS volume
2. Creating a 2x2 GlusterFS volume
3. Creating just backend setup on nodes
4. Starting and stopping services on multiple nodes
5. Set/Unset volume options on a volume
6. Installing packages from yum repositories
7. Subscribing to channels using subscription manager
8. Disabling repos

### CODE REVIEW

**Comment by purnimagupta** [comment]
@sac, would you please have time to go through this PR? I've added all the use-cases except the last one (installing packages from repositories) as that is similar to the task six (installing packages from yum repositories). Please let me know if they're different.
I'd need some more details about the differences in that case.

**Comment by purnimagupta** [comment]
@sac  There were originally some merge conflicts that I have now resolved and pushed to the same branch. 
Waiting for your review. Thanks!

File: basic_gdeploy_doc.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Introduction of gdeploy and how to use gdeploy to configure glusterfs
```
Comment (sac) [inline basic_gdeploy_doc.md:1]: Change `of' to to.

Introduction to gdeploy ...

File: docs/source/package_installation.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Create an empty file and give it any arbitrary name. For the purpose of this
+demonstration, let's call our file ``install_packages.conf``. Add the following
+lines to your newly created config file::
+
+
+   # To install package(s):
+   # Make sure you have the appropriate values in all the placeholders shown in this configuration file.
+   # These values are just for demonstartion purposes.
+
+    [yum]prerequistet
```
Comment (sac) [inline docs/source/package_installation.rst]: Can you remove that prerequistet?

What is that?

File: docs/source/subscription.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
+Subscribe to RedHat Subscription Management
+===========================================
+
+The section 'RH-subscription' can be used to register the system to RHSM,
+attach to a pool, enable repos, disable repos, and unregister from RHSM.
+
+
+**Step 1:**
+
+Create an empty file and give it any arbitrary name. For the purpose of this
```
Comment (sac) [inline docs/source/subscription.rst]: Can you remove that create an empty file ... sentence. And keep it like:

Crate a configuration file, which will look like the one show in the example below.

File: docs/source/subscription.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+ # enabled in the system.
+ # 'repos' can also be specified in the same configuration section to
+ # specify which all repos it should subscribe to, at the time of
+ # registration itself.
+ # We can even make subscription-manager attach this system to a subscription
+ # pool at this section itself. Use the option 'pool' for this.
+
+ [RH-subscription]
+ action=register
+ username=bilbobaggins
```
Comment (sac) [inline docs/source/subscription.rst]: username=user@redhat.com

File: docs/source/subscription.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+ password=shire46
+ auto-attach=true
+ disable-repos=yes
```
Comment (sac) [inline docs/source/subscription.rst:36]: Can you remove disable-repos=yes

We do not need it.

File: docs/source/subscription.rst (hunk 4 above)
Comment (sac) [inline docs/source/subscription.rst]: username=<user>@redhat.com

**Pull request merged**